src/MobileDelivery/MobileController.js

```diff
@@ -4,184 +4,196 @@ import MobileStart from './MobileStart';
-                    name: 'Billy Jones',
-                    position: 'Manager',
-                    email: 'billy@jones.com',
-                    phone: '324-321-7665'
-                },
-                notes: 'Only enter if there is a wizard at the front door.',
-                address: {
-                    street1: '318 2nd Ave Extension South',
-                    street2: '',
-                    city: 'Seattle',
-                    state: 'WA',
-                    zipcode: 98104,
-                    officeNo: ''
-                }
+                agency: '',
+                primaryContact: {},
+                notes: '',
+                address: {},
             },
             donatingAgency :{
-                agency: 'Local Point',
-                primaryContact: {
-                    name: 'Amanda Hernandez',
-                    position: 'Head Cook',
-                    email: 'amanda@lp.com',
-                    phone: '205-385-3312'
-                },
-                address: {
-                    street1: '1245 NE Campus Pkwy',
-                    street2: '',
-                    city: 'Seattle',
-                    state: 'WA',
-                    zipcode: 98105,
-                    officeNo: ''
-                }
-            },
-
-            // TODO: add these fields to deliveryObject
-            // TODO: save signatures as image
-            currentDelivery: {
-                temp: '',
-                daSignature: '',
-                raSignature: '',
-                raPrintName: '',
-                timePickedUp: '',
-                timeCompleted: '',
-                deliveryCompleted: false
+                agency: '',
+                primaryContact: {},
+                address: {},
             },
-            step: 0
+            showSummary: false,
         };
-        this.showStep = this.showStep.bind(this);
-        this.showStart = this.showStart.bind(this);
-        this.nextStep = this.nextStep.bind(this);
-        this.saveValues = this.saveValues.bind(this);
+        this.renderView = this.renderView.bind(this);
+        this.toggleShowSummary = this.toggleShowSummary.bind(this);
     }
 
-    showStart(){
-        this.setState({
-            step: 1
+    componentDidMount() {
+        const { uId: umbrellaId, dId: deliveryId} = this.props.match.params;
+        const deliveryDbRefPath = `deliveries/${umbrellaId}/${deliveryId }`;
+        this.setState({ deliveryDbRefPath: deliveryDbRefPath });
+
+        db.ref(deliveryDbRefPath).on('value', (snapshot) => {  // listen data onchange
+            const deliveryData = snapshot.val();
+            this.aggrDelivery(deliveryData);
         });
     }
 
-    nextStep () {
-        this.setState((prevState) => {
-            return { step: prevState.step + 1 };
-        });
+    componentWillUnmount() {
+        db.ref(this.state.deliveryDbRefPath).off();  // detach listener
+    }
+
+    async aggrDelivery(deliveryData) {
```

Comment:
Given the current schema, this all looks great. However, I wonder if we should discuss some schema changes so you don't have to do two separate queries for the donating agency info.

Reply:
Oh yea DA and DA Members are in different tables, which is a design decision made previously for this proj. I guess it'll be fine here since I created two promises and used Promise.all() to resolve both concurrently. Also, it's nice that they are both O(1) lookup without any complex query.

Reply:
I agree, the way you set it up is really awesome. It's just unfortunate that it had to be such a process to extract that data and combine it into the format you wanted. Although I guess that if we format the data to look too much like what the application wants, we'll end up with a super awkward schema and no data independence if the application changes later. So the current setup is probably a good balance :)

Reply:
Yes, I think the way it is now makes sense.  We do want to maintain data independence. It's common to make multiple requests for one component in modern web app/websites.